docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -0,0 +1,1946 @@
+approach is inspired by
+`Protobuf <https://protobuf.dev/programming-guides/proto3/#updating>`__
+and Typescript's ability to ignore `excess
+fields <https://www.typescriptlang.org/docs/handbook/2/objects.html#excess-property-checks>`__
+via ``as``.
+
+Following is a list of places where automatic data upgrades occur:
+
+**Submissions to the ledger-api**
+
+When you submit a command, and do not explicitly specify a package-id to
+use, canton will automatically upgrade the payloads you give to the most
+recent version of the package that is uploaded on the participant. It
+will also use the most recent implementation of any choices you exercise
+directly through the ledger-api. This behavior can be influenced by
+package preference, which will be covered later.
+
+**Fetches/Creates**
+
+Within choices, the package-ids of the payloads we expect for fetches
+and creates are given by the version of the package we depend on. SCU
+will upgrade (or downgrade) the payloads given/stored in the ledger to
+match these versions when used in a choice body.
+
+**Exercises**
+
+Similarly to fetches, the version of the choice implementation that is
+called from a choice body is determined by the package that the choice’s
+package depends on. SCU will automatically upgrade/downgrade the payload
+stored in the ledger to match that of the choice it is calling.
+
+**Consuming clients (such as daml-script, ts/java codegen)**
+
+When clients query the ledger API for contracts, the returned event
+payload format matches the template originally used for generating the
+event (creating a contract/exercising a choice). It is the
+responsibility of these clients to upgrade/downgrade the payloads they
+receive to match what is expected downstream. Daml-script, as well as
+ts/java codegen, will do this for you to match the payload to the
+package versions they were run/built from.
+
+Upgrading Across the Stack
+--------------------------
+
+These are all the components that interact with upgrades as a feature,
+and this is how you as a user should be aware it interacts.
+
+Canton
+~~~~~~
+
+When considering the Canton ledger nodes, only the Canton participant
+node is aware of smart contract upgrading. The Canton domain nodes are
+only concerned with the protocol version which must be minimally 6 for
+allowing connected participants to use upgradable Daml packages.
+
+Below, we provide a brief overview of the interactions with the
+participant node that have been adapted for supporting the smart
+contract upgrading feature starting with Canton 2.9:
+
+-  DAR upload requests go through an additional validation stage to
+   check the contained new packages for upgrade-compatibility with
+   other packages previously uploaded on the participant.
+
+-  Ledger API command submissions can be automatically or explicitly
+   up/downgraded if multiple upgrade-compatible packages exist.
```

Comment:
Why do they need to be upgrade-compatible? Isn't it that just the package-name must match the one used in the submission? And that submissions with package-names instead of package-ids can be used?

Reply:
cc: @tudor-da

Reply:
I was referring here to >= LF 1.16 packages as LF 1.15 packages do not partake in the package-name resolution:
- [ ] Make it clear that `upgrade-compatible` here refers to LF 1.16 vs LF 1.15 packages

Reply:
Are you going to make this change to the docs?

Reply:
Addressed in 14db8aa380ac21ecf4bb02e862c0abb818b29b9d